### PR TITLE
Remove set -e and add to the end FAILED message

### PR DESCRIPTION
**Pull request by phracek**
Remove set -e and add to the end FAILED message in case test failed.

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -6,12 +6,28 @@
 # TEST_OPENSHIFT_MODE - If set, run OpenShift tests (if present)
 # VERSIONS - Must be set to a list with possible versions (subdirectories)
 
-set -eE
+set -E
```
Comment (zmiklank) [inline test.sh:9]: `-E` is not needed if `-e` is not set IIUC.

File: test.sh [hunk 2]
```diff
@@ -56,9 +74,16 @@ for dir in ${VERSIONS}; do
...
       echo "-> Upstream tests are not present, skipping"
     fi
   fi
   popd > /dev/null
 done
+
+if [[ "$FAILED_VERSIONS" != "" ]]; then
+    echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
+    echo "Test for image ${IMAGE_NAME} FAILED in these versions ${FAILED_VERSIONS}."
+    echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
```
Comment (zmiklank) [inline test.sh:88]: We should exit the script with non-zero value, if some test fails.

File: test.sh [hunk 3]
```diff
@@ -56,9 +74,16 @@ for dir in ${VERSIONS}; do
...
       VERSION=$dir test/run-upstream
+      failed_version "$?" "$dir"
     else
       echo "-> Upstream tests are not present, skipping"
     fi
   fi
   popd > /dev/null
 done
+
+if [[ "$FAILED_VERSIONS" != "" ]]; then
```
Comment (zmiklank) [inline test.sh:85]:
```suggestion
if [[ -n "$FAILED_VERSIONS" ]]; then
```

**Review by zmiklank — CHANGES_REQUESTED** [review]
Thanks for the PR, looks good overall, but I've added few comments.

**Comment by zmiklank** [comment]
And please, check the shellcheck output.

File: test.sh (hunk 1 above)
Comment (phracek) [reply test.sh]: https://github.com/sclorg/container-common-scripts/pull/369/commits/fee1f6a17fc7867f6d7e9fb3d3e36c2cf1340edc

File: test.sh (hunk 2 above)
Comment (phracek) [reply test.sh:86]: https://github.com/sclorg/container-common-scripts/pull/369/commits/fee1f6a17fc7867f6d7e9fb3d3e36c2cf1340edc

**Comment by phracek** [comment]
[test]

File: test.sh [hunk 4]
```diff
@@ -6,12 +6,26 @@
 # TEST_OPENSHIFT_MODE - If set, run OpenShift tests (if present)
 # VERSIONS - Must be set to a list with possible versions (subdirectories)
 
-set -eE
-
 trap 'echo "errexit on line $LINENO, $0" >&2' ERR
```
Comment (zmiklank) [inline test.sh:9]: Also this line no longer has any effect.

**Comment by zmiklank** [comment]
However, I think this PR does not really solve the problem you want to solve. That would require to remove exit on error from build.sh and tag.sh. However, that is not safe to do, as all of testsuites on our container repos rely on that.
And if any image is not even buildable - as it the case in this PR - I see no reason to try to run rest of the tests anyway, even for other versions, as unbuildable image is a big problem by itself.

**Comment by phracek** [comment]
> However, I think this PR does not really solve the problem you want to solve. That would require to remove exit on error from build.sh and tag.sh. However, that is not safe to do, as all of testsuites on our container repos rely on that. And if any image is not even buildable - as it the case in this PR - I see no reason to try to run rest of the tests anyway, even for other versions, as unbuildable image is a big problem by itself.

Currently, in case of `build.sh` and `tag.gs` failed then test is not run anymore. Because in both script there is present `set -e` where it should be.
In case one test has failed, then the next version has to be tested.
This pull request does not belong to building and tagging. Only for tests.

File: test.sh (hunk 4 above)
Comment (phracek) [reply test.sh]: https://github.com/sclorg/container-common-scripts/pull/369/commits/2229acbe720e89bae02bb70256af4efca888e5d7

**Review by zmiklank — APPROVED** [review]

**Pull request merged**